### DISCUSSION TITLE
Why should we split the store into separate slices

### DISCUSSION THREAD

**Discussion by majidsajadi** [pmndrs/zustand#895] [Q&A]
Ok im just started using the `zustand` as my app global state manger and i have a question. i hope i didnt miss anything. 

according to [this wiki](https://github.com/pmndrs/zustand/wiki/Splitting-the-store-into-separate-slices) spliting store into seperate slices is a best practice but since `zustand` is a providerless state manager why should we have a flux-like single store? specially in feature-based architecture. 
for example we have this app

```bash
src
├── featureOne
│   ├── featureOneComponent.tsx
│   └── featureOneStore.ts
├── featureOne
│   ├── featureOneComponent.tsx
│   └── featureOneStore.ts
├── App.tsx
└── index.tsx
```

isnt that a better solution than this one?

```bash
src
├── featureOne
│   ├── featureOneComponent.tsx
│   └── featureOneSlice.ts
├── featureOne
│   ├── featureOneComponent.tsx
│   └── featureOneSlice.ts
├── store.ts
├── App.tsx
└── index.tsx
```

**Comment by dai-shi** [comment]
If you have two stores, and want to create a derived value from the two stores (in a selector function), it will be a bit difficult (or tricky).
So, my general recommendation for multiple store use cases is when those stores are isolated.
